File: Golang/ValidParentheses/ValidParentheses_test.go
package ValidParentheses

import "testing"

func TestIsValid(t *testing.T) {

	if isValid("()") {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isValid("()[]{}") {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isValid("(") == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isValid("(]") == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if isValid("([)]") == false {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/ValidParenthesesTest.py
from unittest import TestCase

from ValidParentheses import ValidParentheses


class TestValidParentheses(TestCase):
    def test_isValid(self):
        vp = ValidParentheses()

        self.assertTrue(vp.isValid("()"))

        self.assertTrue(vp.isValid("()[]{}"))

        self.assertFalse(vp.isValid("("))

        self.assertFalse(vp.isValid("(]"))

        self.assertFalse(vp.isValid("([)]"))
